app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -352,6 +352,97 @@ protected function getQuoteById($quoteId)
         return $immutableQuote;
     }
 
+    /**
+     * Called after pre-auth order is confirmed as authorized on Bolt.
+     *
+     * @param string $orderIncrementId  customer facing order id
+     * @param object $payload           payload sent from Bolt
+     *
+     * @throws Mage_Core_Exception if there is a problem retrieving the bolt transaction reference from the payload
+     */
+    public function receiveOrder( $orderIncrementId, $payload ) {
+        /** @var Mage_Sales_Model_Order $order */
+        $order = Mage::getModel('sales/order')->loadByIncrementId($orderIncrementId);
+        $payloadObject = json_decode($payload);
+        $immutableQuote = $this->getQuoteFromOrder($order);
+
+        Mage::dispatchEvent('bolt_boltpay_order_received_before', array('order'=>$order, 'payload' => $payloadObject));
```

Comment:
As I said in https://github.com/BoltApp/bolt-magento1/pull/396
1) please document these events in README
2) I'm against adding events unless they are absolute necessary

Reply:
> 1.) please document these events in README
As you said, that README will be done as part of PR #396.  This PR, #390, doe not handle that

> 2.) I'm against adding events unless they are absolute necessary
Great.  To move away from core code overrides like `activateOrder` and `setBoltUserId` which we will not be giving developers the opportunity to do, we must give as much entry points at key events as possible and being generous in allowing a means to gracefully update behavior in a forward moving compatible way is preferred to limiting extend-ability by providing no means update at the method level.